### PR TITLE
add difference of between the first value and the last value for aggregators

### PR DESCRIPTION
**Pull request by c61811**
hello，please add this func for opentsdb，that used to calculate difference of between the first value and the last value in the data points for aggregators.
thanks!

### CODE REVIEW

**Comment by c61811** [comment]
please give more advice someone.  this change is there a problem?

**Comment by c61811** [comment]
I hope that the maintainer to seriously consider this feature, because in our scenario, the use of a metric values will continue to increase, but as we look at the index in every changes over a period of time. 

thanks!

**Comment by johann8384** [comment]
This seems like it would be very similar to the "rate" functionality, what is this doing that I have overlooked?

**Comment by c61811** [comment]
image
For example, the AppsSubmitted metric has been growing in Hadoop, we're going to see the specific changes of data of every time.

**Comment by johann8384** [comment]
Isn't this what the "rate" functionality does? Converts a monotonically increasing counter to the derivative?

**Comment by c61811** [comment]
Otherwise, The Int type of uid.cache-hit overflow, I hope to merge into the master branch.

image

**Comment by c61811** [comment]
Isn't the Rate functionality, Although very similar, but the Diff functionality is more intuitive to us.

File: src/uid/UniqueId.java [hunk 1]
```diff
@@ -249,6 +251,17 @@ public void dropCaches() {
   }
 
   /**
+   * Due to the var cache_hits type is int, but the max of int is 2147483648, and int happen spilling
+   */
+  private void reNumCache() {
```
Comment (manolama) [inline src/uid/UniqueId.java]: I'd rather just change the hits to a long instead of rolling over this low.

Comment (c61811) [reply src/uid/UniqueId.java]: hello @manolama, Please pay attention to the tsd.uid.ids-available is minus #1123, How to fix the ids-available that is positive? 
thanks!

**Comment by manolama** [comment]
It does make sense to use the rate mostly, particularly since a bucket may wind up wit a single value. But some folks might like this. Merged in 26355ac18a2982f17edf773ab7ba711f085ac024

**Pull request closed without merging**